src/app/series-import-status-card/series-import-status-card.component.ts

```diff
@@ -0,0 +1,142 @@
+import { Observable } from 'rxjs/Observable';
+import { Subscription } from 'rxjs/Subscription.js'
+import { Component, Input } from '@angular/core';
+import { SeriesImportModel } from '../shared';
+import { HalDoc } from '../core';
+
+import 'rxjs/add/observable/interval';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/mergeMap';
+import 'rxjs/add/operator/takeWhile';
+
+@Component({
+  templateUrl: 'series-import-status-card.component.html',
+  styleUrls: ['./series-import-status-card.component.css'],
+  selector: 'series-import-status-card'
+})
+
+export class SeriesImportStatusCardComponent {
+
+  @Input() seriesImport: SeriesImportModel;
```

Comment:
Making a separate private variable would make me less uncomfortable. I think I also have a different mindset from the way publish was mostly built where components are responsible for so much. Generally I think of a component that takes Inputs as more of a presentational component and some parent component should be responsible for passing down data and handling keeping that data up to date or whatever just passing down changes, and if the child component needs to change the Input, it should be through an Output rather than changing it directly. That's also just not the way Publish was built for the most part, sigh. But that doesn't prevent us from doing better when we know better! Let me see what I can whip up to try to out doing this reload in a Service...

Reply:
Github is broken at the moment, so I'm commenting here until the dust settles.

So I started down the road of modifying feat/podcast-imports service:
https://github.com/PRX/publish.prx.org/commits/feat/podcast-imports-service

I was able to `concat` the two observables, but had a problem with the code in the interval pipe. Turns out you need to subscribe to an observable (and associated `pipe(...)` operators) for the interval code to kick off. Makes sense in hindsight.

So a couple problems:

First, is that we still need to poll for the associated episode imports. The import status card needs that data e.g. to display story titles.

So that means we need to extract the episode import state handling into the new series import service.

Second, we need to kill the poller if we navigate away from the status tab. This mean interacting with the poller on component unmount.

Taking a look at that now.